File: Samples/HtmlToPdf/HtmlToPdfWindowsForms/VB.NET/Form1.vb
Imports BitMiracle.Docotic.Pdf.HtmlToPdf

Public Class Form1
    <STAThread()>
    Public Shared Sub Main()
        Application.SetHighDpiMode(HighDpiMode.SystemAware)
        Application.EnableVisualStyles()
        Application.SetCompatibleTextRenderingDefault(False)

        Dim form As New Form1
        form.Setup()

        Application.Run(form)
    End Sub

    Private Sub Setup()
        textBoxUrl.Text = "https://bitmiracle.com"
        SetConvertingMode(False)
    End Sub

    Private Sub SetConvertingMode(convertingMode As Boolean)
        textBoxUrl.Enabled = Not convertingMode
        buttonConvert.Enabled = Not convertingMode
        progressBarConverting.Visible = convertingMode
    End Sub

    Private Async Sub ButtonConvert_Click(sender As Object, e As EventArgs) Handles buttonConvert.Click
        SetConvertingMode(True)

        Try
            Await ConvertUrlToPdfAsync(textBoxUrl.Text, "HtmlToPdfWindowsForms.pdf")
        Catch ex As HtmlConverterException
            MessageBox.Show(ex.Message, "Conversion failed")
            SetConvertingMode(False)
        End Try
    End Sub

    Private Async Function ConvertUrlToPdfAsync(urlString As String, pdfFileName As String) As Task
        Using converter = Await HtmlConverter.CreateAsync()

            Using pdf = Await converter.CreatePdfAsync(New Uri(urlString))
                pdf.Save(pdfFileName)
            End Using
        End Using

        SetConvertingMode(False)
        MessageBox.Show(Me, $"The output is located in {Environment.CurrentDirectory}")

        Process.Start(New ProcessStartInfo(pdfFileName) With {.UseShellExecute = True})
    End Function
End Class

File: Samples/HtmlToPdf/HtmlToPdfWindowsForms/C#/Form1.cs
using System;
using System.Diagnostics;
using System.Threading.Tasks;
using System.Windows.Forms;

using BitMiracle.Docotic;
using BitMiracle.Docotic.Pdf.HtmlToPdf;

namespace HtmlToPdfWindowsForms
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            textBoxUrl.Text = "https://bitmiracle.com";
            SetConvertingMode(false);
        }

        private void SetConvertingMode(bool convertingMode)
        {
            textBoxUrl.Enabled = !convertingMode;
            buttonConvert.Enabled = !convertingMode;
            progressBarConverting.Visible = convertingMode;
        }

        private async void ButtonConvert_Click(object sender, EventArgs e)
        {
            SetConvertingMode(true);

            try
            {
                await ConvertUrlToPdfAsync(textBoxUrl.Text, "HtmlToPdfWindowsForms.pdf");
            }
            catch (HtmlConverterException ex)
            {
                MessageBox.Show(ex.Message, "Conversion failed");
                SetConvertingMode(false);
            }
        }

        private async Task ConvertUrlToPdfAsync(string urlString, string pdfFileName)
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            using (var converter = await HtmlConverter.CreateAsync())
            {
                using var pdf = await converter.CreatePdfAsync(new Uri(urlString));
                pdf.Save(pdfFileName);
            }

            SetConvertingMode(false);

            MessageBox.Show(this, $"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(pdfFileName) { UseShellExecute = true });
        }
    }
}
